server/server.go

```diff
@@ -179,24 +179,29 @@ func (s *server) buildInvokeChain() {
 
 // RegisterService should not be called by users directly.
 func (s *server) RegisterService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) error {
-	if s.svcInfo != nil {
-		panic(fmt.Sprintf("Service[%s] is already defined", s.svcInfo.ServiceName))
+	if svcInfo == nil {
+		return errors.New("svcInfo is nil. please specify non-nil svcInfo")
 	}
-	s.svcInfo = svcInfo
-	s.handler = handler
-	diagnosis.RegisterProbeFunc(s.opt.DebugService, diagnosis.ServiceInfoKey, diagnosis.WrapAsProbeFunc(s.svcInfo))
+	if handler == nil || reflect.ValueOf(handler).IsNil() {
```

Comment:
If you don't judge whether handler is a pointer type and call IsNil directly, it will panic.

Reply:
I think there's no need to judge IsNil(), because the previous logic also didn't do so.

Reply:
@jayantxie I think IsNil() checks if the handler is a pointer or not and also the previous logic does in check().

Reply:
OK